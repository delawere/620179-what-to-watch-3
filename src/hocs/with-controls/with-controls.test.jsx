import React, {forwardRef} from "react";
import renderer from "react-test-renderer";
import {func, oneOfType, shape, object} from 'prop-types';
import withControls from './with-controls.jsx';

const MockComponent = () => <video />;

const MockComponentWithRef = forwardRef((props, ref) => {
  return <MockComponent {...props} forwardedRef={ref} />;
});

MockComponentWithRef.displayName = `test`;

MockComponent.propTypes = {
  forwardedRef: oneOfType([func, shape({current: object})]),
};

const MockComponentWrapped = withControls(MockComponentWithRef);

it(`withControls renders correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped setActivePlayer={() => {}}/>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
