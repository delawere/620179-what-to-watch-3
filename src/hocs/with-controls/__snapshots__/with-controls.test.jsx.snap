// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withControls renders correctly 1`] = `<video />`;
